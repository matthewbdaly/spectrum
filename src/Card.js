import React, { Component } from 'react';
import './Card.scss';

class Card extends Component {
  render() {
    return (
      <section className="card">
        <section className="header">
          <strong>{this.props.header}</strong>
        </section>
        <section className="body">
          {this.props.children}
        </section>
      </section>
    );
  }
}

export default Card;
