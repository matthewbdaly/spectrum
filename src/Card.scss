@import "variables";

section.card {
    border-style: solid;
    border-width: 2px;
    border-color: lightgray;
    width: 100%;
    padding: 20px;
    margin: 20px;

    .header {
        width: 100%;
        text-align: center;
    }
}
