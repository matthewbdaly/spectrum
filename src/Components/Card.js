// @flow
import React, { Component } from 'react';
import type { Node } from 'react';
import './Card.scss';

type Props = {
  header?: Node,
  children?: Node
}

class Card extends Component<Props> {
  render() {
    return (
      <section className="card">
        <section className="header">
          <strong>{this.props.header}</strong>
        </section>
        <section className="body">
          {this.props.children}
        </section>
      </section>
    );
  }
}

export default Card;
