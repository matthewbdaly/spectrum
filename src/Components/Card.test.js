import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Card />, div);
});

it('should match the snapshot', () => {
  const div = document.createElement('div');
  const el = ReactDOM.render(<Card />, div);
  expect(el).toMatchSnapshot();
});
