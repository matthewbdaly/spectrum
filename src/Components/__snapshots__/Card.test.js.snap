// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
Card {
  "_reactInternalFiber": FiberNode {
    "_debugID": 11,
    "_debugIsCurrentlyTiming": false,
    "_debugOwner": null,
    "_debugSource": Object {
      "fileName": "<PROJECT_ROOT>/src/Components/Card.test.js",
      "lineNumber": 12,
    },
    "alternate": null,
    "child": FiberNode {
      "_debugID": 12,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": [Circular],
      "_debugSource": Object {
        "fileName": "<PROJECT_ROOT>/src/Components/Card.js",
        "lineNumber": 14,
      },
      "alternate": null,
      "child": FiberNode {
        "_debugID": 13,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": [Circular],
        "_debugSource": Object {
          "fileName": "<PROJECT_ROOT>/src/Components/Card.js",
          "lineNumber": 15,
        },
        "alternate": null,
        "child": FiberNode {
          "_debugID": 15,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": [Circular],
          "_debugSource": Object {
            "fileName": "<PROJECT_ROOT>/src/Components/Card.js",
            "lineNumber": 16,
          },
          "alternate": null,
          "child": null,
          "effectTag": 0,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 0,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "children": undefined,
          },
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "children": undefined,
          },
          "ref": null,
          "return": [Circular],
          "sibling": null,
          "stateNode": <strong />,
          "tag": 5,
          "type": "strong",
          "updateQueue": null,
        },
        "effectTag": 0,
        "expirationTime": 0,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": Object {
          "children": <strong />,
          "className": "header",
        },
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": Object {
          "children": <strong />,
          "className": "header",
        },
        "ref": null,
        "return": [Circular],
        "sibling": FiberNode {
          "_debugID": 14,
          "_debugIsCurrentlyTiming": false,
          "_debugOwner": [Circular],
          "_debugSource": Object {
            "fileName": "<PROJECT_ROOT>/src/Components/Card.js",
            "lineNumber": 18,
          },
          "alternate": null,
          "child": null,
          "effectTag": 0,
          "expirationTime": 0,
          "firstEffect": null,
          "index": 1,
          "key": null,
          "lastEffect": null,
          "memoizedProps": Object {
            "children": undefined,
            "className": "body",
          },
          "memoizedState": null,
          "mode": 0,
          "nextEffect": null,
          "pendingProps": Object {
            "children": undefined,
            "className": "body",
          },
          "ref": null,
          "return": [Circular],
          "sibling": null,
          "stateNode": <section
            class="body"
/>,
          "tag": 5,
          "type": "section",
          "updateQueue": null,
        },
        "stateNode": <section
          class="header"
>
          <strong />
</section>,
        "tag": 5,
        "type": "section",
        "updateQueue": null,
      },
      "effectTag": 0,
      "expirationTime": 0,
      "firstEffect": null,
      "index": 0,
      "key": null,
      "lastEffect": null,
      "memoizedProps": Object {
        "children": Array [
          <section
            className="header"
>
            <strong />
</section>,
          <section
            className="body"
/>,
        ],
        "className": "card",
      },
      "memoizedState": null,
      "mode": 0,
      "nextEffect": null,
      "pendingProps": Object {
        "children": Array [
          <section
            className="header"
>
            <strong />
</section>,
          <section
            className="body"
/>,
        ],
        "className": "card",
      },
      "ref": null,
      "return": [Circular],
      "sibling": null,
      "stateNode": <section
        class="card"
>
        <section
                class="header"
        >
                <strong />
        </section>
        <section
                class="body"
        />
</section>,
      "tag": 5,
      "type": "section",
      "updateQueue": null,
    },
    "effectTag": 1,
    "expirationTime": 0,
    "firstEffect": null,
    "index": 0,
    "key": null,
    "lastEffect": null,
    "memoizedProps": Object {},
    "memoizedState": null,
    "mode": 0,
    "nextEffect": null,
    "pendingProps": Object {},
    "ref": null,
    "return": FiberNode {
      "_debugID": 9,
      "_debugIsCurrentlyTiming": false,
      "_debugOwner": null,
      "_debugSource": null,
      "alternate": FiberNode {
        "_debugID": 9,
        "_debugIsCurrentlyTiming": false,
        "_debugOwner": null,
        "_debugSource": null,
        "alternate": [Circular],
        "child": null,
        "effectTag": 0,
        "expirationTime": 1,
        "firstEffect": null,
        "index": 0,
        "key": null,
        "lastEffect": null,
        "memoizedProps": null,
        "memoizedState": null,
        "mode": 0,
        "nextEffect": null,
        "pendingProps": null,
        "ref": null,
        "return": null,
        "sibling": null,
        "stateNode": Object {
          "containerInfo": <div>
            <section
                        class="card"
            >
                        <section
                                    class="header"
                        >
                                    <strong />
                        </section>
                        <section
                                    class="body"
                        />
            </section>
</div>,
          "context": Object {},
          "current": [Circular],
          "finishedWork": null,
          "firstBatch": null,
          "hydrate": false,
          "nextScheduledRoot": null,
          "pendingChildren": null,
          "pendingCommitExpirationTime": 0,
          "pendingContext": null,
          "remainingExpirationTime": 0,
        },
        "tag": 3,
        "type": null,
        "updateQueue": Object {
          "baseState": null,
          "callbackList": null,
          "capturedValues": null,
          "expirationTime": 1,
          "first": Object {
            "callback": null,
            "capturedValue": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "partialState": Object {
              "element": <Card />,
            },
          },
          "hasForceUpdate": false,
          "isInitialized": false,
          "isProcessing": false,
          "last": Object {
            "callback": null,
            "capturedValue": null,
            "expirationTime": 1,
            "isForced": false,
            "isReplace": false,
            "next": null,
            "partialState": Object {
              "element": <Card />,
            },
          },
        },
      },
      "child": [Circular],
      "effectTag": 32,
      "expirationTime": 0,
      "firstEffect": [Circular],
      "index": 0,
      "key": null,
      "lastEffect": [Circular],
      "memoizedProps": null,
      "memoizedState": Object {
        "element": <Card />,
      },
      "mode": 0,
      "nextEffect": null,
      "pendingProps": null,
      "ref": null,
      "return": null,
      "sibling": null,
      "stateNode": Object {
        "containerInfo": <div>
          <section
                    class="card"
          >
                    <section
                              class="header"
                    >
                              <strong />
                    </section>
                    <section
                              class="body"
                    />
          </section>
</div>,
        "context": Object {},
        "current": [Circular],
        "finishedWork": null,
        "firstBatch": null,
        "hydrate": false,
        "nextScheduledRoot": null,
        "pendingChildren": null,
        "pendingCommitExpirationTime": 0,
        "pendingContext": null,
        "remainingExpirationTime": 0,
      },
      "tag": 3,
      "type": null,
      "updateQueue": Object {
        "baseState": Object {
          "element": <Card />,
        },
        "callbackList": null,
        "capturedValues": null,
        "expirationTime": 0,
        "first": null,
        "hasForceUpdate": false,
        "isInitialized": true,
        "isProcessing": false,
        "last": null,
      },
    },
    "sibling": null,
    "stateNode": [Circular],
    "tag": 2,
    "type": [Function],
    "updateQueue": null,
  },
  "_reactInternalInstance": Object {},
  "context": Object {},
  "props": Object {},
  "refs": Object {},
  "state": null,
  "updater": Object {
    "enqueueForceUpdate": [Function],
    "enqueueReplaceState": [Function],
    "enqueueSetState": [Function],
    "isMounted": [Function],
  },
}
`;
